src/shared/sol-fbp-parser.c

```diff
@@ -596,9 +601,15 @@ parse_conn_stmt(struct sol_fbp_parser *p)
 
         sol_fbp_graph_add_in_port(p->graph, dst, dst_port_name, in_port_position);
 
-        r = sol_fbp_graph_add_conn(p->graph, src, src_port_name, src_port_idx, dst, dst_port_name, dst_port_idx, conn_position);
+        if ((long)(int)src_port_idx != src_port_idx)
+            return false;
+
+        if ((long)(int)dst_port_idx != dst_port_idx)
+            return false;
+
+        r = sol_fbp_graph_add_conn(p->graph, src, src_port_name, (int)src_port_idx, dst, dst_port_name, (int)dst_port_idx, conn_position);
         if (r < 0)
-            return handle_conn_error(p, src, &src_port_name, src_port_idx, dst, &dst_port_name, dst_port_idx, &conn_position, r);
+            return handle_conn_error(p, src, &src_port_name, (int)src_port_idx, dst, &dst_port_name, (int)dst_port_idx, &conn_position, r);
```

Comment:
Here is the overflow check

+        if ((long)(int)src_port_idx != src_port_idx)
+            return false;

and 

+        if ((long)(int)src_port_idx != src_port_idx)
+            return false;